### ISSUE TITLE
[PR#5160] minio crashes for 'minio server http://node{1...4}/tmp/disk{1...4}'

### ISSUE THREAD

**Issue by balamurugana** [minio/minio#5520]
Running command `minio server http://node{1...4}/tmp/disk{1...4}` crashes
```

panic: runtime error: index out of range

goroutine 1 [running]:
github.com/minio/minio/cmd.initFormatXL(0xc42034b620, 0x4, 0x4, 0x1, 0x10, 0x4, 0xc4201ac300, 0x4)
	/home/bala/golang/gopath/src/github.com/minio/minio/cmd/format-xl.go:533 +0x410
github.com/minio/minio/cmd.waitForFormatXL(0x11fdc01, 0xc42034b620, 0x4, 0x4, 0x1, 0x10, 0x0, 0x0, 0x0)
	/home/bala/golang/gopath/src/github.com/minio/minio/cmd/prepare-storage.go:119 +0x66d
github.com/minio/minio/cmd.newObjectLayer(0xc42034b620, 0x4, 0x4, 0x2, 0xc42018a690, 0x0, 0xc4200e8ea0)
	/home/bala/golang/gopath/src/github.com/minio/minio/cmd/server-main.go:276 +0x76
github.com/minio/minio/cmd.serverMain(0xc420230c60)
	/home/bala/golang/gopath/src/github.com/minio/minio/cmd/server-main.go:244 +0x7e6
github.com/minio/minio/vendor/github.com/minio/cli.HandleAction(0xf403e0, 0x1166840, 0xc420230c60, 0xc4201c1300, 0x0)
	/home/bala/golang/gopath/src/github.com/minio/minio/vendor/github.com/minio/cli/app.go:499 +0x7c
github.com/minio/minio/vendor/github.com/minio/cli.Command.Run(0x111d97e, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1137531, 0x1c, 0x0, ...)
	/home/bala/golang/gopath/src/github.com/minio/minio/vendor/github.com/minio/cli/command.go:214 +0xa99
github.com/minio/minio/vendor/github.com/minio/cli.(*App).Run(0xc4201bfba0, 0xc420080060, 0x3, 0x3, 0x0, 0x0)
	/home/bala/golang/gopath/src/github.com/minio/minio/vendor/github.com/minio/cli/app.go:260 +0x758
github.com/minio/minio/cmd.Main(0xc420080060, 0x3, 0x3)
	/home/bala/golang/gopath/src/github.com/minio/minio/cmd/main.go:149 +0x8d
main.main()
	/home/bala/golang/gopath/src/github.com/minio/minio/main.go:71 +0x108
```

Here, node1, node2, node3 and node4 are names to four different IPs on same machine.

**Comment by balamurugana** [comment]
It also fails for any input in this pattern e.g. `minio server http://node{1...4}/tmp/disk{1...8}`

**Comment by harshavardhana** [comment]
> minio server http://node{1...4}/tmp/disk{1...4}

The problem is again similar to #5522 the `NewEndpointList` would treat all the endpoints as local while it should keep the endpoints as is instead it will overwrite the endpoint list with '4' endpoints instead of 16 which would lead to crash later on due to mismatch between total set count etc.  Fixing this in master would fix this for PR #5160 as well. 

```go
 // All endpoints are pointing to local host
        if len(endpoints) == localEndpointCount {
                // If all endpoints have same port number, then this is XL setup using URL style endpoints.
                if len(localPortSet) == 1 {
                        if len(localServerAddrSet) > 1 {
                                // TODO: Eventhough all endpoints are local, the local host is referred by different IP/name.
                                // eg '172.0.0.1', 'localhost' and 'mylocalhostname' point to same local host.
                                //
                                // In this case, we bind to 0.0.0.0 ie to all interfaces.
                                // The actual way to do is bind to only IPs in uniqueLocalHosts.
                                serverAddr = net.JoinHostPort("", serverAddrPort)
                        }

                        endpointPaths := endpointPathSet.ToSlice()
                        if len(endpointPaths) == 1 {
                                // This would mean that input arguments were pointing to same disk, perhaps all ips are local
                                // as well pointing to same disk.
                                err = fmt.Errorf("path '%s' cannot be served by different address on same server", endpointPaths[0])
                                return serverAddr, endpoints, setupType, err
                        }
                        endpoints, _ = NewEndpointList(endpointPaths...)
                        setupType = XLSetupType
                        return serverAddr, endpoints, setupType, nil
                }

                // Even though all endpoints are local, but those endpoints use different ports.
                // This means it is DistXL setup.
        }
```

**Comment by harshavardhana** [comment]
https://github.com/minio/minio/pull/5523 - this PR should fix the issue for PR #5160 as well.

**Comment by harshavardhana** [comment]
fixed in PR #5160 rebased after #5523 

```
minio server http://node{1...4}/tmp/disk{1...4} 
[2018-02-15T09:27:45.017441242Z] [FATAL] Invalid command line arguments server=‘:9000’, args=[http://node{1...4}/tmp/disk{1...4}] (path '/tmp/disk1' cannot be served by different address on same server)
```
